### PR TITLE
Use index to complete attribute names.

### PR DESCRIPTION
**Pull request by przepompownia**
#2060

### CODE REVIEW

File: lib/Extension/LanguageServerReferenceFinder/Model/Highlighter.php [hunk 1]
```diff
@@ -66,7 +64,6 @@ public function highlightsFor(string $source, ByteOffset $offset): Promise
                     $highlight->kind
                 );
             }
-            dump('done');
```
Comment (dantleech) [inline lib/Extension/LanguageServerReferenceFinder/Model/Highlighter.php:69]: removed in master now :roll_eyes:

File: lib/Indexer/Adapter/Php/FileSearchIndex.php [hunk 2]
```diff
@@ -56,7 +57,12 @@ public function search(Criteria $criteria): Generator
     public function write(Record $record): void
     {
         $this->open();
-        $info = [$record->recordType(), $record->identifier(), $record instanceof ClassRecord ? $record->type() : null];
+        $info = [
+            $record->recordType(),
+            $record->identifier(),
+            $record instanceof ClassRecord ? $record->type() : null,
+            $record instanceof ClassRecord ? $record->flags() : null,
```
Comment (dantleech) [inline lib/Indexer/Adapter/Php/FileSearchIndex.php:64]: could generalize `flags` to the `Record` class I guess?

Comment (przepompownia) [reply lib/Indexer/Adapter/Php/FileSearchIndex.php:64]: `Record` is an interface. I am not sure if it needs `flags` method. Let's do it in a separate PR or please make a change here.

**Comment by przepompownia** [comment]
Although I removed draft flag, I don't know at the moment if and where to add a test of the new filter.

File: lib/Indexer/Model/Query/Criteria.php [hunk 3]
```diff
@@ -76,6 +77,11 @@ public static function isClassEnum(): IsClassType
         return new IsClassType(ClassRecord::TYPE_ENUM);
     }
 
+    public static function hasClassAttributeFlag(): HasClassFlag
```
Comment (dantleech) [inline lib/Indexer/Model/Query/Criteria.php:80]: maybe just `isAttribute` ?

File: lib/Indexer/Adapter/Php/FileSearchIndex.php (hunk 2 above)
Comment (dantleech) [reply lib/Indexer/Adapter/Php/FileSearchIndex.php:64]: or possible a `HasFlags` interface, but maybe premature it's fine.

**Comment by dantleech** [comment]
@przepompownia I think the test can be here? `lib/Indexer/Tests/Adapter/ReferenceFinder/IndexedNameSearcherTest.php` ?

**Comment by przepompownia** [comment]
> @przepompownia I think the test can be here? `lib/Indexer/Tests/Adapter/ReferenceFinder/IndexedNameSearcherTest.php` ?

I simply did not check if a class I changed has its test case...

File: lib/Extension/LanguageServer/Listener/SelfDestructListener.php [hunk 4]
```diff
@@ -33,7 +33,7 @@ public function handleShutdown(WillShutdown $willShutdown): void
         asyncCall(function () {
             yield delay($this->selfDestructTimeout);
             throw new ExitSession(sprintf(
-                'Waited "%s" seconds after shutdown request for exit notification but did not get one so I\'m self destructing.',
+                'Waited "%s" milliseconds after shutdown request for exit notification but did not get one so I\'m self destructing.',
```
Comment (mamazu) [inline lib/Extension/LanguageServer/Listener/SelfDestructListener.php:36]: Good catch. If you want you could also update the documentation to show the units. (LanguageServerExtension.php:151)

Comment (dantleech) [reply lib/Extension/LanguageServer/Listener/SelfDestructListener.php:36]: I'll update the CL and fix this, thanks @przepompownia !

Comment (dantleech) [reply lib/Extension/LanguageServer/Listener/SelfDestructListener.php:36]: (and @mamazu for the ground work!)

Comment (przepompownia) [reply lib/Extension/LanguageServer/Listener/SelfDestructListener.php:36]: Thanks to both of you too!

**Pull request merged**